### DISCUSSION TITLE
Crons Jobs running as each deployment

### DISCUSSION THREAD

**Discussion by Jirw** [vercel/next.js#58190] [Help]
### Link to the code that reproduces this issue (it's an useless codesandbox i can't provide a reproductible example)

https://codesandbox.io/p/sandbox/github/vercel/next.js/tree/canary/examples/reproduction-template

### To Reproduce

Simply launch a new deployment by pushing on main branch

### Current vs. Expected behavior

Good evening, I'm opening a new topic about crons jobs, which is becoming increasingly problematic.

At the very beginning, the problem was that the cron was launched at each new deployment, whereas the trigger point was only once a month.

By going directly to Vercel, I deactivated manually the Crons functionality to simply disable all activity about the cron but same problem, I deleted the vercel.json file (which contained the crons config), same problem, it runs at each deployment.

Recently, I added a new api route in the same subfolder where the first cron was config:

folder
->route 1 (initially declared in vercel.json)
->route 2 (not declared in anything)

Every time I deploy, route 2 is also called up, I don't know what to do, it's all a mess...

I've tried deleting Vercel's deployment cache.
Currently the Cron functionality is disabled on Vercel, I push a vercel.json = {}, nothing does my 2 routes are called at each deployment.

Thank you for reading and for your precious help...

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
"react": "18.2.0",
"react-dom": "^18.2.0",
"next": "13.4.19",
"node": 18.x
```

### Which area(s) are affected? (Select all that apply)

Not sure

**Comment by timneutkens** [comment]
Converted this to discussions as it doesn't seem to relate to Next.js but instead your question is about Vercel. I'll forward it to the right people but people use https://github.com/orgs/vercel/discussions instead for Vercel related questions 👍

**Comment by AndyBitz** [comment]
@Jirw If this is related to cron jobs on Vercel, then https://github.com/orgs/vercel/discussions would indeed be a more suitable place.

That said, am I understanding you correct: You have a cron job on your project, that you've disabled on the Dashboard, and even deleted from `vercel.json`, but it is still invoked? In order to investigate this further, we'll need the ID of the affected Vercel project. You can find it in your project settings on the Vercel Dashboard, under  the General tab.

**Reply by Jirw** [reply]
Hello, thank you for your reply. 
Can I continue to answer on this post?
Yes, you've understood the problem, a route is even called even though I didn't mention it in vercel.json or anywhere else in the project.

the id is: prj_CxVsH4RiGCEhUJq6EWaKIOuhKtls

Thanks for your time.

**Reply by AndyBitz** [reply]
Thanks, I've looked at the cron job invocations of this project and couldn't find any invocation for at least the last 4 weeks.

Furthermore, our system doesn't have any cron job for this project at the moment, so they were deleted successfully when you removed your `vercel.json` file.

This makes me think that the request that you see come from elsewhere. Generally, you can go to the "Usage" tab on the Vercel Dashboard, there you can see the number of invocations from your cron jobs for any given time and project.

To investigate the origin of your requests further, I'd recommend you to check the "Logs" tab on your Project. Requests that originated from cron jobs will have the `user-agent: vercel-cron/1.0` as header. If that's not the case, then the requests are made by someone else.

And as a side-note, anyone could set this user-agent header, so it's not a guarantee that the request was made by Vercel. If you need authenticity, you can see how to do this [here in our docs](https://vercel.com/docs/cron-jobs/manage-cron-jobs#securing-cron-jobs).

**Comment by fernando24j** [comment]
I can share some light about what's is going on. 

Is related with Next.js, same happened to me, but on a docker container with other service of cronjob. The reason is you declared the route as a 'GET', so depending the logic inside the route the next builder recognizes the route as a static page, so tries to compile it.

### Solution 

You should use other API verbs such as 'POST or 'PUT as the target of your cron job

**Reply by Jirw** [reply]
Hey ! Interesting solution, my routes are both 'GET', i will try to change the API verbs. 
Thank's mate !

**Reply by kevinmanase** [reply]
This allows you to deploy the project successfully but it won't actually work. Here's how it work

> To trigger a cron job, Vercel makes an HTTP GET request to your project's production deployment URL, using the path provided in your project's vercel.json file. An example endpoint Vercel would make a request to in order to trigger a cron job might be: https://*.vercel.app/api/cron.

Hence, you will get an error 405 [Method not allowed]

[Here is what actually helped](https://dev.to/chrisnowicki/how-to-create-a-vercel-cron-job-using-nextjs-13-2amb)

tldr;
```js
// Nextjs route segment config
export const dynamic = 'force-dynamic' // Force dynamic (server) route instead of static page

export async function GET() {
 ...your cron job code here
}
```

**Reply by Jirw** [reply]
Even better ! Thank you very much !

**Comment by zineanteoh** [comment]
> This allows you to deploy the project successfully but it won't actually work. Here's how it work
> 
> > To trigger a cron job, Vercel makes an HTTP GET request to your project's production deployment URL, using the path provided in your project's vercel.json file. An example endpoint Vercel would make a request to in order to trigger a cron job might be: https://*.vercel.app/api/cron.
> 
> Hence, you will get an error 405 [Method not allowed]
> 
> [Here is what actually helped](https://dev.to/chrisnowicki/how-to-create-a-vercel-cron-job-using-nextjs-13-2amb)
> 
> tldr;
> 
> ```js
> // Nextjs route segment config
> export const dynamic = 'force-dynamic' // Force dynamic (server) route instead of static page
> 
> export async function GET() {
>  ...your cron job code here
> }
> ```

was having problem promoting a vercel build to production because a cron route was mistakenly rendered as a static route. 

this helped me resolve the issue -- thank you!
